ploggy/base/handler.py

```diff
@@ -18,7 +78,14 @@ class Handler:
     pipe: IO = stderr
 
     def log(self, entry: Entry) -> None:
-        print(self.format(entry), file=self.pipe, flush=True)
+        try:
+            fn, lno, func, _ = find_caller()
+        except ValueError:
+            fn, lno, func = "(unknown file)", 0, "(unknown function)"
+
+        line_ctx = f"{fn}:{lno}:{func}"
+
+        print(self.format(line_ctx, entry), file=self.pipe, flush=True)
 
-    def format(self, entry: Entry) -> str:
+    def format(self, line_ctx: str, entry: Entry) -> str:
```

Comment:
breaking the internal .format syntax

Reply:
this is alright since it actually gets the right caller